File: Solution in CPP/Que6.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

string longestCommonPrefix(vector<string>& strs) {
    if(strs.empty()){
        return ""; // If the vector is empty, return an empty string
    } 

    for(int i = 0; i<strs[0].size();++i) {
        for(int j = 1;j<strs.size();++j) {
            if(i >= strs[j].size() || strs[j][i] != strs[0][i]) {
                return strs[0].substr(0, i);
            }
        }
    }
    // If no mismatch is found, return the first string
    return strs[0];
}

int main() {
    vector<string> strings={"flower", "flow", "flight"};
    cout<<"Longest common prefix: " << longestCommonPrefix(strings) << endl;
    return 0;
}

File: Solution in RUST/Que6.rs
// Implement a function that finds the longest common prefix of a given set of strings.


fn longest_common_prefix(strs: &[String]) -> String {
    if strs.is_empty() {
        return String::new(); // If the vector is empty, return an empty string
    }

    for (i, &ch) in strs[0].as_bytes().iter().enumerate() {
        for other_str in &strs[1..] {
            if i >= other_str.len() || other_str.as_bytes()[i] != ch {
                return strs[0][..i].to_string();
            }
        }
    }
    // If no mismatch is found, return the first string
    strs[0].clone()
}

fn main() {
    let strings = vec!["flower".to_string(), "flow".to_string(), "flight".to_string()];
    println!("Longest common prefix: {}", longest_common_prefix(&strings));
}
